src/core/token/ERC721Core.sol

```diff
@@ -149,8 +180,44 @@ contract ERC721Core is ERC721AQueryable, Core, Multicallable {
      *  @param quantity The quantity of tokens to mint.
      *  @param data ABI encoded data to pass to the beforeMint hook.
      */
-    function mint(address to, uint256 quantity, bytes calldata data) external payable {
-        _beforeMint(to, _nextTokenId(), quantity, data);
+    function mint(address to, uint256 quantity, string calldata baseURI, bytes calldata data) external payable {
+        uint256 tokenId = _nextTokenId();
+        if (bytes(baseURI).length > 0) {
+            _updateMetadata(to, tokenId, quantity, baseURI);
+        }
+        _beforeMint(to, tokenId, quantity, data);
+        _safeMint(to, quantity, "");
+    }
+
+    /**
+     *  @notice Mints a token with a signature. Calls the beforeMint hook.
+     *  @dev Reverts if beforeMint hook is absent or unsuccessful.
+     *  @param to The address to mint the token to.
+     *  @param quantity The quantity of tokens to mint.
+     *  @param data ABI encoded data to pass to the beforeMint hook.
+     *  @param signature The signature produced from signing the minting request.
+     */
+    function mintWithSignature(
+        address to,
+        uint256 quantity,
+        string calldata baseURI,
+        bytes calldata data,
+        bytes memory signature
+    ) external payable {
+        address signer = _hashTypedData(
+            keccak256(abi.encode(TYPEHASH_SIGNATURE_MINT_ERC721, to, quantity, keccak256(bytes(baseURI)), data))
```

Comment:
@jakeloo @kumaryash90 do we need to keccak the data field here too? I believe we do but can't remember